.github/actions/do_build_ock_artefact/action.yml

```diff
@@ -0,0 +1,71 @@
+name: build-ock-artefacts
+description: Action to build the oneapi-construction-kit as an artefact
+
+# Note we need to support llvm tip at some point
+
+inputs:
+  llvm_version:
+    description: 'llvm version we want to use (18-19)'
+    default: '19'
+  target:
+    description: 'target architecture'
+
+# TODO: This has not been tested yet on windows so would likely need some updating.
+runs:
+  # We don't want a new docker just a list of steps, so mark as composite
+  using: "composite"
+  steps:
+      - name: calc vars
+        id: calc_vars
+        uses: ./.github/actions/calc_vars
+        with:
+          target: ${{ inputs.target }}
+
+      - name: print vars
+        shell: bash
+        run: |
+          echo arch = ${{steps.calc_vars.outputs.arch}}
+          echo toolchain = ${{steps.calc_vars.outputs.toolchain}}
+
+      # installs tools, ninja, installs llvm and sets up sccache
+      - name: setup
+        uses:  ./.github/actions/setup_build
+        with:
+          llvm_version: ${{ inputs.llvm_version }}
+          llvm_build_type: RelAssert
+          cross_arch: ${{ steps.calc_vars.outputs.arch == 'x86_64' && '' || steps.calc_vars.outputs.arch }}
+
+      - name: build ock x86
+        if: steps.calc_vars.outputs.arch  == 'x86_64'
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_targets: install
+          offline_kernel_tests: OFF
+          extra_flags: -DCA_ENABLE_TESTS=OFF -DCA_ENABLE_EXAMPLES=OFF -DCA_ENABLE_DOCUMENTATION=OFF
+
+      - name: build ock other ${{ matrix.target }}
+        if: steps.calc_vars.outputs.arch  != 'x86_64'
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_targets: install
+          toolchain_file: ${{ steps.calc_vars.outputs.toolchain }}
+          extra_flags: -DCA_BUILTINS_TOOLS_DIR=${{ github.workspace }}/llvm_install_native/bin -DCA_ENABLE_TESTS=OFF -DCA_ENABLE_EXAMPLES=OFF -DCA_ENABLE_DOCUMENTATION=OFF
+          # Do we need the offline kernel as an artefact? If so currently this requires an external clc or qemu to be installed.
+          offline_kernel_tests: OFF
+          host_fp16: ON
+
+      # Prune it as there is too much things in there we don't want to use
+      # We may want to extend this a bit as time goes on such as clc or UnitCL
```

Comment:
We should probably add CMake options that prevent things from being installed, rather than installing them and then removing them.

Reply:
I did do that above as much as I could see, I do agree we should have some more options to do this.